Add the git commit hash (and if there are local uncommited changes) t…

eleenest opened the pull request:
…o the vendor command we added (for a more detailed version command).

example output:
$ probe._link.vendor(19, [0x00])[1:64].tostring()
b'udb_0.0.1_1_11c0525bb00ed38b59e42612df9653d52b82a339'

eleenest committed:
Add the git commit hash (and if there are local uncommited changes) to the vendor command we added (for a more detailed version command).

example output:
$ probe._link.vendor(19, [0x00])[1:64].tostring()
b'udb_0.0.1_1_11c0525bb00ed38b59e42612df9653d52b82a339'

gaborcsapo commented:
Looks good, this will be useful for debugging other people's version. 

One idea: can we change the "0" "1" to "" "-modified" and apply it as a postfix? This way it's more apparent to developers who don't dig into this part of the source.

eleenest committed:
Address review comments.
Make the udb_version a c symbol, so it can be conveniently seen / updated in the makefile (and in the yaml files in the future), next to the Daplink version.
Note: 2 levels of macro's needed to "stringize" the symbol, per https://gcc.gnu.org/onlinedocs/cpp/Stringizing.html.  Perhaps move these to a header file if we use this again.

gaborcsapo commented on projectfiles/make_gcc_arm/stm32h743ii_if/Makefile:
```diff
@@ -53,6 +53,7 @@ LD_SCRIPT = $(OUT_DIR)/$(TARGET).generated$(LINKER_EXT)
 
 CC_SYMBOLS = \
     -DDAPLINK_VERSION=254 \
+    -DUDB_VERSION="0.9.0" \
```
Note: We'll have to make sure this is also managed with Progen

gaborcsapo reviewed (approved):
Note: We'll have to make sure this is also managed with Progen
